compiler/expr_visitor.py

```diff
@@ -379,6 +379,10 @@ def visit_Yield(self, node):
     self.writer.write('return {}, nil'.format(value.expr))
     self.writer.write_label(resume_label)
     result = self.block.alloc_temp()
+    self.writer.write(textwrap.dedent("""\
```

Comment:
I think you want to do:

```
if πRaised != nil {
	πE = πRaised
	πRaised = nil
	if πE != nil {
		continue
	}
}
```

You could also merge the write() call below into this:

```
self.writer.write_tmpl(textwrap.dedent("""\
    if πRaised != nil {
    \tπE = πRaised
    \tπRaised = nil
    \tif πE != nil {
    \t\tcontinue
    \t}
    }
    $result = πSent"""), result=result.name)
```